tern/extensions/scancode/executor.py

```diff
@@ -114,13 +114,17 @@ def add_file_data(layer_obj, collected_files):
 
 class Scancode(Executor):
     '''Execute scancode'''
-    def execute(self, image_obj):
+    def execute(self, image_obj, redo=False):
         '''Execution should be:
             scancode -ilpcu --quiet --json - /path/to/directory
         '''
         for layer in image_obj.layers:
             # load the layers from cache
-            common.load_from_cache(layer)
+            common.load_from_cache(layer, redo)
+            # reading files again in case of redo
+            if redo:
+                logger.debug('Reading files in filesystem...')
+                layer.add_files()
             if not layer.files_analyzed:
                 # the layer doesn't have analyzed files, so run analysis
                 file_list = collect_layer_data(layer)
```

Comment:
Here is where the redo check needs to happen `if redo or not layer.files_analyzed:`